### PR TITLE
MAINT: upgrade anaconda and ga workflow versions

### PR DESCRIPTION
**Pull request by mmcky**
This PR

- [x] upgrades to `anaconda=2024.02`
- [x] upgrades GitHub action workflow versions to latest

### CODE REVIEW

**Comment by mmcky** [comment]
- [ ] `ERROR: No matching distribution found for interpolation==2.2.4` is causing an issue with the newer version of `numba==0.59`

**Comment by doctor-phil** [comment]
@mmcky Numba is not used directly in the notes as far as I know, so we can probably get away with removing the pinned version from the environment

File: environment.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ channels:
   - default
 dependencies:
   - python=3.9
-  - anaconda=2022.10
+  - anaconda=2024.02
```
Comment (mmcky) [inline environment.yml:6]: looks like the `pip` install fails when specifying `2024.02` so maybe we can revert this to update the `GitHub actions` infrastructure

**Comment by doctor-phil** [comment]
```
ERROR: No matching distribution found for interpolation==2.2.4
```

**Comment by mmcky** [comment]
@doctor-phil let's wait on https://github.com/EconForge/interpolation.py/pull/112 and then re-run with a new pin

**Comment by mmcky** [comment]
@doctor-phil it looks like the pip pins are too much for the new `2024` anaconda distribution -- probably too many downgrades. I will leave this open but let me know if I can help in any way.

**Comment by mmcky** [comment]
@doctor-phil is there a reason for such a restrictive build environment? Can we relax the version pins?

**Comment by doctor-phil** [comment]
@mmcky we can relax some of them I think, these are the pins that I put in place in #225 to avoid the constant build failures due to third-party package upgrades.

**Comment by mmcky** [comment]
Closing this as merged in #265

**Pull request closed without merging**